src/fvmcts/action_coordination/varel.jl

```diff
@@ -226,23 +226,51 @@ function update_statistics!(mdp::JointMDP{S,A}, tree::JointMCTSTree{S,A,VarElSta
         # RECOVER local action corresp. to ucb action
         # TODO: Review this carefully. Need @req for action index for agent.
         local_action = [ucb_action[c] for c in comp]
-        local_action_idxs = [get_agent_actionindex(mdp, c, a) for (a, c) in zip(local_action, comp)]
+        local_action_idxs = [agent_actionindex(mdp, c, a) for (a, c) in zip(local_action, comp)]
 
         comp_ac_idx = LinearIndices(comp_tup)[local_action_idxs...]
 
         # NOTE: NOW we can update stats. Could generalize incremental update more here
         lock(tree.lock) do
             tree.coordination_stats.n_component_stats[s][idx][comp_ac_idx] += 1
-            q_comp_value = sum(q[c] for c in comp)
+            q_comp_value = sum(q[c] for c in comp) # TODO!!!
             tree.coordination_stats.q_component_stats[s][idx][comp_ac_idx] +=
                 (q_comp_value - tree.coordination_stats.q_component_stats[s][idx][comp_ac_idx]) / tree.coordination_stats.n_component_stats[s][idx][comp_ac_idx]
         end
     end
 end
 
+# TODO: is this the correct thing to do? 
+# My guess is no, but not sure.
+function update_statistics!(mdp::JointMDP{S,A}, tree::FVMCTSTree{S,A,VarElStatistics{S}},
+    s::S, ucb_action::A, q::Float64) where {S, A}
+    n_agents = length(s)
+
+    for (idx, comp) in enumerate(tree.coordination_stats.coord_graph_components)
+
+        # Create cartesian index tuple
+        comp_tup = Tuple(1:length(tree.all_agent_actions[c]) for c in comp)
+
+        # RECOVER local action corresp. to ucb action
+        # TODO: Review this carefully. Need @req for action index for agent.
+        local_action = [ucb_action[c] for c in comp]
+        local_action_idxs = [agent_actionindex(mdp, c, a) for (a, c) in zip(local_action, comp)]
+
+        comp_ac_idx = LinearIndices(comp_tup)[local_action_idxs...]
+
+        # NOTE: NOW we can update stats. Could generalize incremental update more here
+        lock(tree.lock) do
+            tree.coordination_stats.n_component_stats[s][idx][comp_ac_idx] += 1
+            q_comp_value = q # TODO!!!
```

Comment:
I see. The TODO here is warranted haha. The equivalent for the `q::AbstractVector` case should be
`q_comp_value = q * length(comp)` ?

Reply:
Because if we had a global reward vector where all q values were the same, then the line `q_comp_value = sum(q[c] for c in comp)` would yield the same result right?

Reply:
Yeah, that's the thing I wanted to confirm and seems alright.